<?xml version="1.0" ?>
<launch>

  <arg name="cpp" default="true"/>
  <arg name="bluefox" default="false"/>
  <arg name="flea3" default="true"/>
  <arg name="openni" default="false"/>

  <arg name="image_topic" default="/mv_29900372/image_raw" if="$(arg bluefox)"/>

  <!-- <arg name="image_topic" default="/pg_13344889/image_raw" if="$(arg flea3)"/> -->

  <!-- <arg name="image_topic" default="/pg_13344889/image_rect_color" if="$(arg flea3)"/>  -->
  <arg name="image_topic" default="/pg_13344889/image_raw" if="$(arg flea3)"/>

  <arg name="image_topic" default="/camera/rgb/image_rect_color" if="$(arg openni)"/>


  <!-- Desired height in cm -->
  <arg name="des_height_cm" default="2.4"/>

  <!--options: L,R,M,A -->
  <!-- Left now, as robot pours from the left, hence M,R are in jeporady -->
  <arg name="vis_side" default="-selection L"/>

  <!-- Runs the camera drivers -->
  <include file="$(find queen_cam_calibration)/launch/kinect_calib.launch">
    <arg name="bluefox" value="$(arg bluefox)"/>
    <arg name="flea3" value="$(arg flea3)"/>
  </include>

  <!-- Runs the Apriltag node -->
  <include file="$(find liquid_apriltags)/launch/new_cylinder_tag_detection.launch">
    <arg name="image" value="$(arg image_topic)"/>
  </include>

<arg name="receiving_cup_glass" default="true"/> <!--This defaults to glass recieving beaker, other option is just plastic beaker reciever -->



  <node name='Liquid_Level_Detector_Node' pkg='liquid_level_detection' type='pouring_w_april_main.py' args="$(arg vis_side)" output="screen" unless="$(arg cpp)">
      <remap from="image" to="$(arg image_topic)"/>
  </node>
<group if="$(arg receiving_cup_glass)">
  <node name='Level_Detector_Node_cpp' pkg='liquid_level_detection' type='level_detect_node' output="screen" if="$(arg cpp)" >
    <param name="desired_height_cm" type="double" value="$(arg des_height_cm)"/>
    <param name="apriltag_cut_area" type="double" value="0.025"/>

    <param name="apriltag_size" type="double" value="0.019"/>
    <param name="cylinder_height" type="double" value="0.089"/>
    <param name="x_offset_pix" type="int" value="50"/>
    <param name="y_offset_pix" type="int" value="20"/>
    <param name="debug" type="bool" value="false"/>
    <remap from="~image" to="$(arg image_topic)"/>
    <remap from="~detected_height" to="LDHeight"/>
  </node>
</group>

<group unless="$(arg receiving_cup_glass)">
  <node name='Level_Detector_Node_cpp' pkg='liquid_level_detection' type='level_detect_node' output="screen" if="$(arg cpp)" >
    <param name="desired_height_cm" type="double" value="$(arg des_height_cm)"/>
    <param name="apriltag_cut_area" type="double" value="0.025"/>

    <param name="apriltag_size" type="double" value="0.0192"/>
    <param name="cylinder_height" type="double" value="0.0755"/>
    <param name="x_offset_pix" type="int" value="50"/>
    <param name="y_offset_pix" type="int" value="20"/>
    <param name="debug" type="bool" value="false"/>
    <remap from="~image" to="$(arg image_topic)"/>
    <remap from="~detected_height" to="LDHeight"/>
  </node>
</group>




  <!--height for glass cylinder 0.089m tagsize: 0.017, yoffset was 40 but that was too low, area 41.6666-->
  <!--Front tag0 was 0.02m and height of 0.091 in front -->
  <!-- New plastic reciever height is 7.55cm, tagsize = 0.0192cm, Area: 38.515cm**2 , yoffset was 20, xoffset 50 (still got jumpts)-->


</launch>
